src/components/ButtonDropdown.vue

```diff
@@ -10,7 +10,7 @@
       </span>
     </button>
     <div v-show="isOpen" class="button-dropdown-content" :style="dropdownStyle">
-      <slot></slot>
+      <slot v-if="isOpen"></slot>
```

Comment:
I think `v-show` might preserve the state on open/close, IIRC that is just doing an inline `display: none` toggle type of thing.